liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/server/StartDebugMojoSupport.java

```diff
@@ -214,11 +214,13 @@ protected void runLibertyMojoDeploy(boolean forceLooseApp) throws MojoExecutionE
     }
 
     protected void runLibertyMojoInstallFeature(Element features) throws MojoExecutionException {
-        Xpp3Dom config = ExecuteMojoUtil.getPluginGoalConfig(getLibertyPlugin(), "install-feature", log);;
-        if (features != null) {
-            config = Xpp3Dom.mergeXpp3Dom(configuration(features), config);
+        if (!project.getProperties().containsKey("container")) {  // for now, container mode does not support installing features
```

Comment:
I don't understand doing this here. Why are you not checking for container mode in   `public void libertyInstallFeature()` in the DevMojo goal which is what you implied should be done when you changed ci.common?

Reply:
libertyInstallFeature() calls runLibertyMojoInstallFeature().  There are other places in DevMojo where it calls runLibertyMojoInstallFeature() directly (since libertyInstallFeature() and runLibertyMojoInstallFeature() have different method signatures e.g. arguments and exceptions).  So instead of checking the container flag before every call to runLibertyMojoInstallFeature(), we just check it in runLibertyMojoInstallFeature().

Reply:
IMO that seems like a design flaw.

Reply:
runLibertyMojoInstallFeature() used to be within DevMojo, but was refactored out.  We could have another wrapper method within DevMojo but that seems excessive.